src/transformer/events/mod_quiz/question_answered/essay.php

```diff
@@ -24,7 +24,7 @@ function essay(array $config, \stdClass $event, \stdClass $questionattempt, \std
     $repo = $config['repo'];
     $user = $repo->read_record_by_id('user', $event->relateduserid);
     $course = $repo->read_record_by_id('course', $event->courseid);
-    $attempt = $repo->read_record_by_id('quiz_attempts', $questionattempt->questionusageid);
+    $attempt = $repo->read_record('quiz_attempts', array('uniqueid' => $questionattempt->questionusageid));
```

Comment:
Making this suggestion to keep for consistency with other code.
```suggestion
    $attempt = $repo->read_record('quiz_attempts', ['uniqueid' => $questionattempt->questionusageid]);
```